ntoskrnl/ke/amd64/kiinit.c

```diff
@@ -285,14 +276,32 @@ VOID
 KiInitializeP0BootStructures(
     _Inout_ PLOADER_PARAMETER_BLOCK LoaderBlock)
 {
+    KDESCRIPTOR GdtDescriptor = {{0},0,0}, IdtDescriptor = {{0},0,0};
+    PKGDTENTRY64 TssEntry;
+    PKTSS64 TssBase;
+
     /* Set the initial stack, idle thread and process for processor 0 */
     LoaderBlock->KernelStack = (ULONG_PTR)KiP0BootStack;
     LoaderBlock->Thread = (ULONG_PTR)&KiInitialThread;
     LoaderBlock->Process = (ULONG_PTR)&KiInitialProcess.Pcb;
     LoaderBlock->Prcb = (ULONG_PTR)&KiInitialPcr.Prcb;
 
+    /* Get GDT and IDT descriptors */
+    __sgdt(&GdtDescriptor.Limit);
+    __sidt(&IdtDescriptor.Limit);
+
+    /* Get the boot TSS from the GDT */
+    TssEntry = KiGetGdtEntry(GdtDescriptor.Base, KGDT64_SYS_TSS);
+    TssBase = KiGetGdtDescriptorBase(TssEntry);
```

Comment:
Hmm... Why moving those out of the initPcr function? Are they going to be used elsewhere?

Reply:
The TssBase is passed to `KiInitializeProcessorBootStructures` by the code that launches the other processors. Only for the boot processor we get this from the values that the bootloader has set up.

Reply:
> The TssBase is passed to KiInitializeProcessorBootStructures by the code that launches the other processors.

Not sure where that other code (for the other processors) also calls KiInitializeProcessorBootStructures :eyes:

Reply:
ah, this explains why :joy: